src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/RazorLanguageServerCapability.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Newtonsoft.Json.Linq;
+using OmniSharp.Extensions.LanguageServer.Protocol.Server.Capabilities;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer
+{
+    internal record RazorLanguageServerCapability(bool LanguageQuery, bool RangeMapping, bool EditMapping, bool MonitorProjectConfigurationFilePath)
+    {
+        private const string RazorCapabilityKey = "razor";
+        private static readonly RazorLanguageServerCapability Default = new RazorLanguageServerCapability(LanguageQuery: true, RangeMapping: true, EditMapping: true, MonitorProjectConfigurationFilePath: true);
+
+        public static void AddTo(ServerCapabilities capabilities)
+        {
+            // We have to use the experimental capabilities bucket here because not all platforms maintain custom capabilities. For instance
+            // in Visual Studio scenarios it will deserialize serve capabilties into what it believes is "valid" and then will re-pass said
```

Comment:
```suggestion
            // in Visual Studio scenarios it will deserialize server capabilities into what it believes is "valid" and then will re-pass said
```